pil-analyzer/tests/parse_display.rs

```diff
@@ -719,16 +719,57 @@ namespace T(8);
 
 #[test]
 fn trait_def() {
-    let input = "trait Add<T: Add, Q> {
+    let input = "trait Add<T, Q> {
         add: T, T -> Q,
     }
 ";
 
-    let expected = "    trait Add <T: Add, Q> {
+    let expected = "    trait Add<T, Q> {
         add: T, T -> Q,
     }
 ";
 
     let analyzed = analyze_string::<GoldilocksField>(input);
     assert_eq!(expected, analyzed.to_string())
 }
+
+#[test]
+fn array_type_trait() {
+    let input = "trait ArraySum<T> {
+        array_sum: T[5] -> T,
```

Comment:
That's not what I meant. It should be something like `4 + 1`, something that is not just a number.